app/src/main/java/io/homeassistant/companion/android/notifications/MessagingService.kt

```diff
@@ -50,11 +51,23 @@ class MessagingService : FirebaseMessagingService() {
         }
 
         remoteMessage.notification?.let {
-            Log.d(TAG, "Message Notification: ${it.title} -> ${it.body}")
-            sendNotification(it.title, it.body!!)
+            if (it.title == "request_location_update") {
+                Log.d(TAG, "Request location update")
+                requestLocationUpdate()
+            } else {
+                Log.d(TAG, "Message Notification: ${it.title} -> ${it.body}")
+                sendNotification(it.title, it.body!!)
+            }
         }
     }
 
+    private fun requestLocationUpdate() {
+        val intent = Intent(this, LocationBroadcastReceiver::class.java)
+        intent.action = LocationBroadcastReceiver.ACTION_REQUEST_LOCATION_UPDATES
```

Comment:
This intent action doesn't request a single high accuracy location update.  If you want that you will need a new action that calls `requestSingleAccurateLocation` and call that from here.